app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,38 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    return false if Setting['oidc_algorithm'].nil? || !OpenSSL::PKey::RSA.new.respond_to?(:set_key)
```

Comment:
Does this mean jwt auth won't work with ruby 2.3, or at least until jwt releases a new version? should the condition be more explicit so we know when we can remove it?

Reply:
Yes, I have done the fix here: https://github.com/jwt/ruby-jwt/pull/333. The gem releases every 3 months but this time they have identified few issues with the gem therefore they might do the release in few weeks. 

Also, I have an eye on the gem as I am working on few issues of the gem. Once, we have the new version of the gem, then I will update this condition. 

What do you think?

Reply:
@tbrisker to be safe I have added a comment that addresses out concern. Does that make sense?

Reply:
I would prefer to raise or at least log an error instead of failing silently in this case - otherwise users have no way of telling why this failed.